src/collective/mustread/tracker.py

```diff
@@ -91,11 +103,117 @@ def most_read(self, days=None, limit=None):
         for record in self.query_all(query):
             yield api.content.get(UID=record.uid)
 
-    def schedule_must_read(self, obj, userids=None, deadline=None):
-        raise NotImplementedError()
+    def schedule_must_read(self, obj, userids, deadline):
+        # get existing must read items for this object
+        qry = self._read(uid=utils.getUID(obj))
+        existing_users = [m.userid for m in qry.all()]
+
+        path = '/'.join(obj.getPhysicalPath())
+        uid = utils.getUID(obj)
+        hostname = utils.hostname()
+        now = datetime.utcnow()
+        current_user = api.user.get_current().id
+        new_users = []
+        for userid in userids:
+            if userid in existing_users:
+                # skip uses that already read the item
+                # or have an open mustread request
+                log.info('user {0} already has read / has to read {1}'.format(
+                    userid, path))
+                continue
+            # @guido: should we change the deadline for existing must_reads?
```

Comment:
Yes that makes sense. However then you should also update the `scheduled_at` (and possibly the `scheduled_by` but see above) and now you're painting yourself into a corner where it's possible that a document was read by a user before it was scheduled (because she read it on the first scheduling, before it got scheduled again). Of course that is already possible anyway: a user may have read an item before it got scheduled at all. Do you have test coverage for that edge case?